### PR TITLE
feat: add multiprocessing dataloading

### PR DESCRIPTION
**Pull request by tadejsv**
- [x] torch
- [x] paddle
- [x] ~~keras~~ - could not get keras to work with passthrough of `workers`, (and `multiprocessing`, etc.). By default Keras already behaves as if it has a (single) background process loading data - leaving it at that for now
- [x] tests (how?)

### CODE REVIEW

File: finetuner/tuner/__init__.py [hunk 1]
```diff
@@ -87,6 +88,10 @@ def fit(
         ``"cpu"`` and ``"cuda"`` (for GPU)
     :param callbacks: A list of callbacks. The progress bar callback
         will be pre-prended to this list.
+    :param num_workers: Number of workers used for loading the data.
+
+        This works onlywith Pytorch and Paddle Paddle, and has no effect when using
```
Comment (davidbp) [inline finetuner/tuner/__init__.py:93]: mispelled `only with`

File: finetuner/tuner/base.py [hunk 2]
```diff
@@ -176,6 +177,10 @@ def fit(
         :param batch_size: The batch size to use for training and evaluation
         :param num_items_per_class: Number of items from a single class to include in
             the batch. Only relevant for class datasets
+        :param num_workers: Number of workers used for loading the data.
+
+            This works onlywith Pytorch and Paddle Paddle, and has no effect when using
```
Comment (davidbp) [inline finetuner/tuner/base.py:182]: same here

**Review by bwanglzu — APPROVED** [review]
LGTM!

**Comment by bwanglzu** [comment]
FYI @gmastrapas

**Pull request merged**